From bad76d43b1c5916e80fb60f9dcc2459b0d576213 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 31 Jul 2014 18:05:50 -0700
Subject: [PATCH 919/974] netdev:octeon-ethernet: Set bypass mode for RGMII
 interface

Parse the device tree for rx-clk-delay-bypass and set bypass mode.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/octeon/ethernet.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/octeon/ethernet.c b/drivers/net/ethernet/octeon/ethernet.c
index d68ecd8..6dd63c2 100644
--- a/drivers/net/ethernet/octeon/ethernet.c
+++ b/drivers/net/ethernet/octeon/ethernet.c
@@ -767,16 +767,24 @@ static int cvm_oct_get_port_status(struct device_node *pip)
 	for (i = 0; i < num_interfaces; i++) {
 		int num_ports = cvmx_helper_interface_enumerate(i);
 		int mode = cvmx_helper_interface_get_mode(i);
+		struct device_node *port_node;
 
 		for (j = 0; j < num_ports; j++) {
+			port_node = cvm_oct_node_for_port(pip, i, j);
 			switch (mode) {
+			case CVMX_HELPER_INTERFACE_MODE_AGL:
+				if (port_node &&
+				    of_get_property(port_node,
+						    "cavium,rx-clk-delay-bypass",
+						    NULL))
+					cvmx_helper_set_agl_rx_clock_delay_bypass(i, j, true);
+					/* Continue on */
 			case CVMX_HELPER_INTERFACE_MODE_RGMII:
 			case CVMX_HELPER_INTERFACE_MODE_GMII:
 			case CVMX_HELPER_INTERFACE_MODE_XAUI:
 			case CVMX_HELPER_INTERFACE_MODE_RXAUI:
 			case CVMX_HELPER_INTERFACE_MODE_SPI:
-			case CVMX_HELPER_INTERFACE_MODE_AGL:
-				if (cvm_oct_node_for_port(pip, i, j) != NULL)
+				if (port_node)
 					cvmx_helper_set_port_valid(i, j, true);
 				else
 					cvmx_helper_set_port_valid(i, j, false);
@@ -786,8 +794,6 @@ static int cvm_oct_get_port_status(struct device_node *pip)
 			case CVMX_HELPER_INTERFACE_MODE_SGMII:
 			case CVMX_HELPER_INTERFACE_MODE_QSGMII:
 			{
-				struct device_node *port_node;
-				port_node = cvm_oct_node_for_port(pip, i, j);
 				if (port_node != NULL)
 					cvmx_helper_set_port_valid(i, j, true);
 				else
-- 
2.6.2

